src/utils.js

```diff
@@ -0,0 +1,35 @@
+import readline from 'readline';
+
+export async function readLineAsync(query) {
+  return new Promise((resolve, reject) => {
+    if (arguments.length !== 1) {
+      reject(new Error('arguments must be 1'));
+    }
+
+    if (typeof query !== 'string') {
+      reject(new Error('query must be string'));
+    }
+
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    rl.question(query, input => {
+      rl.close();
+      resolve(input);
+    });
+  });
+}
+
+export async function retryUntilSuccess(callbackFn) {
+  try {
+    return await callbackFn();
+  } catch {
+    return await retryUntilSuccess(callbackFn);
+  }
+}
```

Comment:
[ 첫번째 입력에 공백이 있는 경우 ]

<img width="464" alt="스크린샷 2025-02-13 오후 11 28 23" src="https://github.com/user-attachments/assets/74e57cb2-7abd-4b34-a6b5-78179d88ec42" />


[ 첫번째 입력에 공백 제거 ]

<img width="478" alt="스크린샷 2025-02-13 오후 11 30 21" src="https://github.com/user-attachments/assets/85f91f31-1031-4b35-bb61-21087ba45ea3" />

이름 입력에 공백이 들어간 경우 에러가 나서 정상적으로 입력이 되지 않습니다. 입력에 공백을 허용하지 않는 것을 어플리케이션이 제한하는 것은 가능하겠지만, 사용자가 왜 방금의 입력이 잘못된 것인지를 인지하지 못하는 것은 문제입니다.

```suggestion
export async function retryUntilSuccess(callbackFn) {
  try {
    return await callbackFn();
  } catch {
    console.log('잘못된 입력입니다! 이유 : ~~~~');
  
    return await retryUntilSuccess(callbackFn);
  }
}
```
위와 같이 오류의 이유를 피드백하는 플로우가 필요해보입니다.

또한 한대의 차 이름만을 명시하는게 가능한 상황입니다. 자동차 '경주' 앱인 만큼 2개 이상의 차가 입력되어야 경주 기능이 동작하도록 예외처리 하는 것이 좋아보입니다.

Reply:
확인해봤습니다! 공백자체를 문자로 인식해서 test1에 띄어쓰기를 하면 공백까지 문자로 인식해서 6자가 넘어 글자수 오류에 걸린거였습니다...!

에러메세지가 있었으면 어떤 오류인지 바로 파악했을것같아서 반성하면서 코드 고치도록 하겠습니다...ㅎㅎ 자동차 갯수도 한개라면 동작하지 않도록 수정하겠습니다. 감사합니다!